consensus/state.go

```diff
@@ -1204,6 +1204,10 @@ func (cs *ConsensusState) finalizeCommit(height int) {
 	err := stateCopy.ApplyBlock(eventCache, cs.proxyAppConn, block, blockParts.Header(), cs.mempool)
 	if err != nil {
 		cs.Logger.Error("Error on ApplyBlock. Did the application crash? Please restart tendermint", "err", err)
```

Comment:
Perhaps we should remove this error message then and instead make it
```go
if err != nil {
    if err := cmn.Kill(); err != nil {
        // Maybe log that we successfully killed Tendermint
    } else {
        cs.Logger.Error("Error on ApplyBlock. Did the application crash? Please restart Tendermint manually", "err", err)
    }
    return
}

Reply:
I think we need to alert the user when the app crashes, and then initiate tear down. If tear down fails, we should log that error too.

Reply:
Alright, at least let's change the message asking them to start Tendermint yet we are doing it right after logging the message
<img width="589" alt="screen shot 2017-10-28 at 3 44 42 pm" src="https://user-images.githubusercontent.com/4898263/32139178-f04e6574-bbf6-11e7-8e52-d17eb8162955.png">

Reply:
What do you mean? We tell the user to restart tendermint, and then we go and kill it. So then they, or their OS, can restart it!

Reply:
Oops I was definitely mistaken. For some reason when I read the commit message `Tendermint should die if app hash does so OS can handle restarts` I assumed that Tendermint would be auto-restarted or the OS would handle it exiting, but I re-read everything after your comment. My apologies.